addons/api/mirage/factories/host-set.js

```diff
@@ -12,4 +12,14 @@ export default factory.extend({
       'remove-hosts',
     ],
   id: (i) => `host-set-id-${i}`,
+  type() {
+    if (this.hostCatalog) {
```

Comment:
What about `return this.hostCatalog?.type || super.type;`?  I'd just double check that `super` actually works here.  Not sure.

Reply:
The super syntax might be something else.  This requires a little research.  Because mirage factories aren't native classes, they may have their own super system.  Hopefully it is super duper.

Reply:
I can reasearch the `super.type` no problem. But your comment got me thinking:
- The factory is extended from `generated/factory` which already has a base type.
- Assuming we have `factory.type` with a value set previously, how about `return this.hostCatalog?.type || factory.type;`

Reply:
Is `factory.type` accessible?  It might be on a different property, not sure.

Reply:
jsut checked, it is accessible: `factory.attrs.type`